Update MathComponent.cpp

2Ez4Joker opened the pull request:
MathComponent changes.

2Ez4Joker committed:
Update MathComponent.cpp

MathComponent changes.

kamawwanai commented on src/Math/MathComponent.cpp:
```diff
@@ -1,12 +1,85 @@
+#include<stdio.h>
 #include <iostream>
 
+double cross_product(Point2f a, Point2f b, Point2f c) {
```
Переменные не говорящие не понятно, что такое a,b,c

kamawwanai commented on src/Math/MathComponent.cpp:
```diff
@@ -1,12 +1,85 @@
 #include <iostream>
 
+double cross_product(Point2f a, Point2f b, Point2f c) {
+	double ax = b.x - a.x;
```
код стайл ax - плохо, a_x - хорошо

kamawwanai commented on src/Math/MathComponent.cpp:
```diff
@@ -1,12 +1,85 @@
     -> Intersection {
-    // алгоритм здесь
-    // заместо этих двух строк
+	for (int i = 0;i < 2;++i)
```
отсутствует форматирование кода во всей этой функции, тяжело в целом смотреть код

kamawwanai replied:
код до сих пор не отформатирован, в vscode для этого нажми сочетание клавиш shift+alt+F

kamawwanai replied:
не забудь поставить для этого расширение clangd и отключить все расширение для C++ от microsoft. Они конфликтуют.

kamawwanai commented on src/Math/MathComponent.cpp:
```diff
@@ -1,12 +1,85 @@
+}
+// Функция для проверки принадлежности точки треугольнику
+bool point_inside_triangle(Point2f a, Point2f b, Point2f c, Point2f p) {
+	double cp1 = cross_product(a, b, p);
```
опять же название переменных (расшифровку cp лучше не искать в интернете)

kamawwanai commented on src/Math/MathComponent.cpp:
```diff
@@ -1,12 +1,85 @@
+			Intersection.push_back(triangle_2[i]);
+		}
+	}
     (void)triangle_1;
```
вот эту строчку надо убрать, она ничего не делает

kamawwanai commented on src/Math/MathComponent.cpp:
```diff
@@ -1,12 +1,85 @@
+		}
+	}
     (void)triangle_1;
     (void)triangle_2;
```
и эту

kamawwanai commented on src/Math/MathComponent.cpp:
```diff
@@ -15,12 +88,21 @@ auto MathComponent::calculate_intersection(Triangle triangle_1,
+	max_y = max(max_y, abs(triangle_1[2].y));
+	max_y = max(max_y, abs(triangle_2[0].y));
+	max_y = max(max_y, abs(triangle_2[1].y));
+	max_y = max(max_y, abs(triangle_2[2].y));
```
Во-первых, не подключено пространство имен std, и не нужно его подключать, поэтому пиши `std::max` и `std::abs`
Во-вторых, для функции `std::max()` необходимо `#include <algorithm>`
В `std::max()` можно передавать initializer_list и не писать так много повторяющихся строчек, например,
`std::max({1.F,2.F,3.F})`

kamawwanai commented on src/Math/MathComponent.cpp:
```diff
@@ -15,12 +88,21 @@ auto MathComponent::calculate_intersection(Triangle triangle_1,
+	max_y = max(max_y, abs(triangle_2[0].y));
+	max_y = max(max_y, abs(triangle_2[1].y));
+	max_y = max(max_y, abs(triangle_2[2].y));
+	float kaef = 400/max(max_y, max_x);
```
название переменных, транслит лучше вообще не использовать.
коэффициент можно перевести как coefficient или ratio.

kamawwanai commented on src/Math/MathComponent.cpp:
```diff
@@ -15,12 +88,21 @@ auto MathComponent::calculate_intersection(Triangle triangle_1,
+	max_y = max(max_y, abs(triangle_2[2].y));
+	float kaef = 400/max(max_y, max_x);
+    
     (void)triangle_1;
```
как и в прошлой функции, эту строчку необходимо убрать, она ничего не делает

kamawwanai commented on src/Math/MathComponent.cpp:
```diff
@@ -15,12 +88,21 @@ auto MathComponent::calculate_intersection(Triangle triangle_1,
+	float kaef = 400/max(max_y, max_x);
+    
     (void)triangle_1;
     (void)triangle_2;
```
эту тоже

kamawwanai commented on src/Math/MathComponent.cpp:
```diff
@@ -1,12 +1,85 @@
+					new_point.x = new_x;
+					new_point.y = new_y;
+					Intersection.push_back(new_point);
+				}
```
Во-первых, можно использовать else if, ведь если indicator стал false на условиях до, то остальные проверки не нужны.
Во-вторых, std::max() и std::min(), подробное объяснение написала в одном из комментариев выше.

kamawwanai reviewed (changes requested):
Поправь, пожалуйста, в соответствие с комментариями

2Ez4Joker committed:
Update MathComponent.cpp

Все поправил по комментам. Еще в функции calculate_intersection ругался на использование Intersection , так что создал новый вектор Intersection_points внутри функции, и вывожу его.

2Ez4Joker committed:
Update MathComponent.cpp

2Ez4Joker committed:
Update MathComponent.cpp

2Ez4Joker committed:
Update MathComponent.cpp

2Ez4Joker reviewed (commented):
Поправил.

kamawwanai commented on src/Math/MathComponent.cpp:
```diff
@@ -1,26 +1,94 @@
+			float first_point_y_2 = triangle_2[j].y;
+			float second_point_x_2 = triangle_2[(j + 1) % 3].x;
+			float second_point_y_2 = triangle_2[(j + 1) % 3].y;
+			if ((first_point_x_1 - second_point_x_1) * (first_point_y_2 - second_point_y_2) - (first_point_y_1 - second_point_y_2) * (first_point_x_2 - second_point_x_2) != 0)
```
также разбей условия на функции с говорящими названиями.
Приятнее читать if (is_on_one_line(point1, point2)) или наподобие. Зависит от того, что твоя функция делает

kamawwanai replied:
это касается и последующих условий

kamawwanai commented on src/Math/MathComponent.cpp:
```diff
@@ -1,26 +1,94 @@
-    return {};
+->Intersection
+{
+	std::vector<Point2f> Intersection_points;
```
для этого используется Intersection, объявленный в <Common.hpp>

kamawwanai reviewed (changes requested):
В целом, хорошо исправил, вот еще немного и будет идеально.

2Ez4Joker committed:
Update MathComponent.cpp

Поправлено по комментам.

2Ez4Joker commented:
Поправлено.

kamawwanai reviewed (approved):
Молодец, спасибо!
![45498d0c94602e5474f70105b670c310.jpg](https://user-images.githubusercontent.com/87859313/235511819-94dcffe2-fefb-4be6-bdfb-f2f473dcfa44.jpg)

2Ez4Joker committed:
Update MathComponent.cpp

2Ez4Joker commented:
Была опечатка в переменной. + Попросили сортировку точек сделать

2Ez4Joker committed:
Update MathComponent.cpp

Опечатка, из-за которой только две точки от каждого треугольника смотрелись а не три.
